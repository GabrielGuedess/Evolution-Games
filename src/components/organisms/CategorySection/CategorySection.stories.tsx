import { Story, Meta } from '@storybook/react/types-6-0';

import CategorySection from './CategorySection';

export default {
  title: 'Organisms/CategorySection',
  component: CategorySection,
} as Meta;

export const Default: Story = () => <CategorySection />;
