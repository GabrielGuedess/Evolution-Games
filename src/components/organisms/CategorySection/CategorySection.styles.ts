import media from 'styled-media-query';

import { Container } from 'components/atoms/Container/Container';

import styled, { css } from 'styled-components';

export const Wrapper = styled.section`
  margin-top: 6rem;
  margin-bottom: 6rem;
`;

export const FirstColumn = styled(Container)`
  ${({ theme }) => css`
    height: 760px;
    max-height: 760px;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin-top: ${theme.spacings.xsmall};
    gap: 2rem;

    ${media.greaterThan('large')`
      height: 500px;
      max-height: initial;
      flex-direction: row;
    `}

    ${media.greaterThan('huge')`
      height: 570px;
      max-height: initial;
      margin-top: ${theme.spacings.large};
    `}
  `}
`;

export const SecondColumn = styled(Container)`
  height: 100%;
  max-height: 370px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 2rem;

  ${media.greaterThan('large')`
    width: auto;
    height: 650px;
    max-height: 650px;
  `}

  ${media.greaterThan('huge')`
    height: 787px;
    max-height: 787px;
  `}
`;

export const ActionImageCategory = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 396px;
    height: 370px;
    flex: 1;
    position: relative;

    .image-category {
      cursor: pointer;
      transition: transform 0.5s ease;
    }

    &:hover {
      h4 {
        text-shadow: 0px 0px 3px ${theme.colors.white};
      }

      .image-category {
        transform: scale(1.2);
      }
    }

    ${media.greaterThan('large')`
      width: 100%;
      max-width: initial;
      height: 100%;
    `}
  `}
`;

export const TerrorImageCategory = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 396px;
    height: 370px;
    flex: 1;
    position: relative;

    .image-category {
      cursor: pointer;
      transition: transform 0.5s ease;
    }

    &:hover {
      h4 {
        text-shadow: 0px 0px 3px ${theme.colors.white};
      }

      .image-category {
        transform: scale(1.2);
      }
    }

    ${media.greaterThan('large')`
      width: 100%;
      max-width: initial;
      height: 100%;
      flex: 1.69;
    `}
  `}
`;

export const QualityImageCategory = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 396px;
    height: 370px;
    position: relative;

    .image-category {
      cursor: pointer;
      transition: transform 0.5s ease;
    }

    &:hover {
      h4 {
        text-shadow: 0px 0px 3px ${theme.colors.white};
      }

      .image-category {
        transform: scale(1.2);
      }
    }

    ${media.greaterThan('large')`
      width: 100%;
      max-width: initial;
      height: 100%;
    `}
  `}
`;

export const ImageHeader = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 35px;
  right: 0;
  left: 0;
  z-index: 1;
`;

export const ImageTitle = styled.strong`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.regular};
    font-style: normal;
    color: ${theme.colors.secondary};
    line-height: ${theme.spacings.xsmall};
    margin-bottom: ${theme.spacings.xxsmall};
    text-align: center;

    ${media.greaterThan('large')`
      font-size: ${theme.font.sizes.xbig};
      margin-bottom: ${theme.spacings.small};
    `}
  `}
`;

export const ImageSubTitle = styled.strong`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxmedium};
    font-weight: ${theme.font.regular};
    font-style: normal;
    color: ${theme.colors.whiteText};
    text-align: center;

    ${media.greaterThan('large')`
      font-size: ${theme.font.sizes.xxbig};
    `}
  `}
`;
