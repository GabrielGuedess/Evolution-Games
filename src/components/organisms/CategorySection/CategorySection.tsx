import Image from 'next/image';
import Link from 'next/link';

import { Container } from 'components/atoms/Container/Container';
import { Title } from 'components/atoms/Title/Title';

import * as S from './CategorySection.styles';

export const CategorySection = () => (
  <S.Wrapper>
    <Container>
      <Title textAlign="left">Categorias</Title>
    </Container>
    <S.FirstColumn>
      <Link href="/">
        <S.ActionImageCategory>
          <S.ImageHeader>
            <S.ImageTitle>Para os amantes da emoção</S.ImageTitle>
            <S.ImageSubTitle>Ação e Aventura</S.ImageSubTitle>
          </S.ImageHeader>
          <Image
            layout="fill"
            loading="lazy"
            quality={100}
            objectFit="cover"
            alt="Ação e Aventura"
            className="image-category"
            src="/img/category/actionAdventure.png"
          />
        </S.ActionImageCategory>
      </Link>
      <Link href="/">
        <S.TerrorImageCategory>
          <S.ImageHeader>
            <S.ImageTitle>Para os amantes do terror</S.ImageTitle>
            <S.ImageSubTitle>Terror</S.ImageSubTitle>
          </S.ImageHeader>
          <Image
            alt="Terror"
            layout="fill"
            quality={100}
            loading="lazy"
            objectFit="cover"
            className="image-category"
            src="/img/category/horror.png"
          />
        </S.TerrorImageCategory>
      </Link>
    </S.FirstColumn>
    <S.SecondColumn>
      <Link href="/">
        <S.QualityImageCategory>
          <S.ImageHeader>
            <S.ImageTitle>Para os amantes da qualidade</S.ImageTitle>
            <S.ImageSubTitle>Jogos em 4k</S.ImageSubTitle>
          </S.ImageHeader>
          <Image
            layout="fill"
            loading="lazy"
            quality={100}
            alt="Jogos 4k"
            objectFit="cover"
            className="image-category"
            src="/img/category/4k.png"
          />
        </S.QualityImageCategory>
      </Link>
    </S.SecondColumn>
  </S.Wrapper>
);
