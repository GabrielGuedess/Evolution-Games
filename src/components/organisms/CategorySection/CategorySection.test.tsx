/* eslint-disable react/jsx-no-constructed-context-values */
import { Context as ResponsiveContext } from 'react-responsive';

import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { CategorySection } from './CategorySection';

type Image = { src: string; alt: string; width: number; height: number };

jest.mock('next/image', () => ({
  __esModule: true,
  default: (image: Image) => (
    <img
      src={image.src}
      alt={image.alt}
      width={image.width}
      height={image.height}
    />
  ),
}));

describe('<CategorySection />', () => {
  it('should render the <CategorySection /> on the desktop', () => {
    const { container } = renderWithTheme(
      <ResponsiveContext.Provider value={{ width: 1170 }}>
        <CategorySection />
      </ResponsiveContext.Provider>,
    );

    // Arrange
    const title = screen.getByRole('heading', { name: 'Categorias' });
    const imageOne = screen.getByRole('img', { name: 'Acão e Aventura' });
    const imageTwo = screen.getByRole('img', { name: 'Terror' });
    const imageThree = screen.getByRole('img', { name: 'Jogos 4k' });

    // Assert
    expect(title).toBeInTheDocument();
    expect(imageOne).toHaveAttribute(
      'src',
      '/img/category/actionAdventure.png',
    );
    expect(imageTwo).toHaveAttribute('src', '/img/category/horror.png');
    expect(imageThree).toHaveAttribute('src', '/img/category/4k.png');

    expect(imageOne).toHaveAttribute('width', '608');
    expect(imageOne).toHaveAttribute('height', '568');

    expect(imageTwo).toHaveAttribute('width', '1029');
    expect(imageTwo).toHaveAttribute('height', '568');

    expect(imageThree).toHaveAttribute('width', '1656');
    expect(imageThree).toHaveAttribute('height', '787');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the <CategorySection /> on the mobile', () => {
    renderWithTheme(
      <ResponsiveContext.Provider value={{ width: 1169 }}>
        <CategorySection />
      </ResponsiveContext.Provider>,
    );

    // Arrange
    const imageOne = screen.getByRole('img', { name: 'Acão e Aventura' });
    const imageTwo = screen.getByRole('img', { name: 'Terror' });
    const imageThree = screen.getByRole('img', { name: 'Jogos 4k' });

    // Assert
    expect(imageOne).toHaveAttribute('width', '396');
    expect(imageOne).toHaveAttribute('height', '370');

    expect(imageTwo).toHaveAttribute('width', '396');
    expect(imageTwo).toHaveAttribute('height', '370');

    expect(imageThree).toHaveAttribute('width', '396');
    expect(imageThree).toHaveAttribute('height', '370');
  });
});
