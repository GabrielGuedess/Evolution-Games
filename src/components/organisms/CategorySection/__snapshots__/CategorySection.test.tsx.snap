// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  '<CategorySection /> should render the <CategorySection /> on the desktop 1'
] = `
.c1 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

.c2 {
  color: #F4F4F5;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 500;
  text-align: left;
  font-size: 1.6rem;
}

.c0 {
  margin-top: 6rem;
  margin-bottom: 6rem;
}

.c3 {
  height: 760px;
  max-height: 760px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 1.6rem;
  gap: 2rem;
}

.c9 {
  height: 100%;
  max-height: 370px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 2rem;
}

.c4 {
  width: 100%;
  max-width: 396px;
  height: 370px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.c4 .image-category {
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
}

.c4:hover h4 {
  text-shadow: 0px 0px 3px #FFFFFF;
}

.c4:hover .image-category {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c8 {
  width: 100%;
  max-width: 396px;
  height: 370px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.c8 .image-category {
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
}

.c8:hover h4 {
  text-shadow: 0px 0px 3px #FFFFFF;
}

.c8:hover .image-category {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c10 {
  width: 100%;
  max-width: 396px;
  height: 370px;
  position: relative;
}

.c10 .image-category {
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
}

.c10:hover h4 {
  text-shadow: 0px 0px 3px #FFFFFF;
}

.c10:hover .image-category {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 35px;
  right: 0;
  left: 0;
  z-index: 1;
}

.c6 {
  font-size: 1.6rem;
  font-weight: 400;
  font-style: normal;
  color: #717B8A;
  line-height: 1.6rem;
  margin-bottom: 0.8rem;
  text-align: center;
}

.c7 {
  font-size: 2.0rem;
  font-weight: 400;
  font-style: normal;
  color: #F4F4F5;
  text-align: center;
}

@media (min-width:1170px) {
  .c1 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c1 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c1 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (min-width:1170px) {
  .c3 {
    height: 500px;
    max-height: initial;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:1440px) {
  .c3 {
    height: 570px;
    max-height: initial;
    margin-top: 4.0rem;
  }
}

@media (min-width:1170px) {
  .c9 {
    width: auto;
    height: 650px;
    max-height: 650px;
  }
}

@media (min-width:1440px) {
  .c9 {
    height: 787px;
    max-height: 787px;
  }
}

@media (min-width:1170px) {
  .c4 {
    width: 100%;
    max-width: initial;
    height: 100%;
  }
}

@media (min-width:1170px) {
  .c8 {
    width: 100%;
    max-width: initial;
    height: 100%;
    -webkit-flex: 1.69;
    -ms-flex: 1.69;
    flex: 1.69;
  }
}

@media (min-width:1170px) {
  .c10 {
    width: 100%;
    max-width: initial;
    height: 100%;
  }
}

@media (min-width:1170px) {
  .c6 {
    font-size: 2.5rem;
    margin-bottom: 2.4rem;
  }
}

@media (min-width:1170px) {
  .c7 {
    font-size: 2.8rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      Categorias
    </h1>
  </div>
  <div
    class="c1 c3"
  >
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <strong
          class="c6"
        >
          Para os amantes da emoção
        </strong>
        <strong
          class="c7"
        >
          Ação e Aventura
        </strong>
      </div>
      <img
        alt="Ação e Aventura"
        src="/img/category/actionAdventure.png"
      />
    </div>
    <div
      class="c8"
    >
      <div
        class="c5"
      >
        <strong
          class="c6"
        >
          Para os amantes do terror
        </strong>
        <strong
          class="c7"
        >
          Terror
        </strong>
      </div>
      <img
        alt="Terror"
        src="/img/category/horror.png"
      />
    </div>
  </div>
  <div
    class="c1 c9"
  >
    <div
      class="c10"
    >
      <div
        class="c5"
      >
        <strong
          class="c6"
        >
          Para os amantes da qualidade
        </strong>
        <strong
          class="c7"
        >
          Jogos em 4k
        </strong>
      </div>
      <img
        alt="Jogos 4k"
        src="/img/category/4k.png"
      />
    </div>
  </div>
</section>
`;
